### DISCUSSION TITLE
Saving RTSP Stream to disk gets timed out over TCP connection.

### DISCUSSION THREAD

**Discussion by ebala96** [bluenviron/mediamtx#758] [Questions]
I've created a Docker image that contains both the RTSP server and FFmpeg [https://github.com/aler9/rtsp-simple-server/discussions/278#discussioncomment-549104](url)
rtsp-simple-server version: **v0.17.13**

I am trying to save the RTSP stream to disk ondemand.
Followed the steps given in https://github.com/aler9/rtsp-simple-server#save-published-videos-to-disk and edited my yml config file
![image](https://user-images.githubusercontent.com/36764710/147753313-64846c20-f1f6-416b-9d98-52f53b622c8c.png)

When I publish the stream using the following FFmpeg command
ffmpeg -i rtsp://mystreamCredsWithIP/stream1 -c copy -f rtsp -rtsp_transport tcp rtsp://localhost:8554/frompublished
and try to play the path for ondemand the stream gets saved for around 8-10 seconds and then gets timed out.

Logs from RTSP container. 
![image](https://user-images.githubusercontent.com/36764710/147754872-929dc113-c719-4f9b-bde1-c12cb8ec983c.png)
![image](https://user-images.githubusercontent.com/36764710/147754917-1acd1157-d8b8-48fb-92c5-90a118a300bd.png)

Upon using ffprobe -rtsp_transport tcp rtsp://localhost:8554/ondemand got the following logs.
![image](https://user-images.githubusercontent.com/36764710/147754214-e1cb65ef-babc-448b-ad14-d64202b8eba8.png)

I am not able to view the stream from VLC as well when trying to compress an RTSP stream using https://github.com/aler9/rtsp-simple-server#remuxing-re-encoding-compression.
The stream gets timed out when I play the stream from the compressed path.
 
But contrastingly when I run the FFmpeg commands directly inside the RTSP container both save and compressing works fine.

I see they mentioned TCP as the protocol for ondemand to work and to disable UDP for that stream when talking to it,
Is this the cause of the timeout issue? Or am I missing anything as part of the config?

**Comment by aler9** [comment]
Hello, `runOnDemand` is not meant to be used to save streams to disk, please [follow instructions](https://github.com/aler9/rtsp-simple-server#save-streams-to-disk) and use either `runOnReady` or `runOnPublish`.

[a comment by an automated account is not shown]
